src/rpc/rawtransaction.cpp

```diff
@@ -1745,101 +1537,296 @@ UniValue signrawtransaction(const UniValue& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid sighash param");
     }
 
-    bool fHashSingle = ((nHashType & ~SIGHASH_ANYONECANPAY) == SIGHASH_SINGLE);
+    if ((txVersion == SC_CERT_VERSION) && (nHashType != SIGHASH_ALL)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Unsupported sighash param for certificate");
+    }
 
-    // Script verification errors
-    UniValue vErrors(UniValue::VARR);
+#ifdef ENABLE_WALLET
```

Comment:
Check the place. It should be after the if in the line 1506: `if (params.size() > 2 && !params[2].isNull()) {`